pkg/api/v1alpha1/cluster.go

```diff
@@ -389,6 +389,13 @@ func validateControlPlaneLabels(clusterConfig *Cluster) error {
 	return nil
 }
 
+func validateControlPlaneEndpoint(clusterConfig *Cluster) error {
+	if clusterConfig.Spec.DatacenterRef.Kind == DockerDatacenterKind && clusterConfig.Spec.ControlPlaneConfiguration.Endpoint != nil && clusterConfig.Spec.ControlPlaneConfiguration.Endpoint.Host != "" {
```

Comment:
If the validation only applies to docker (unfortunate) maybe we should find a different place this. It's an edge case, we never had to run validations at the cluster level that were provider specific 

It's only one for now so I'm ok with this, but can we create a task in the project to come back to this?

Reply:
Yes I'll add a follow up task to the project!